tests/postgres_tests/test_array.py

```diff
@@ -363,6 +363,24 @@ class Meta:
         self.assertEqual(len(errors), 1)
         self.assertEqual(errors[0].id, 'postgres.E002')
 
+    def test_nested_field_checks(self):
+        """
+        #25867 - Nested array fields should set their leaf field's
```

Comment:
Alright, this was mostly a copy of `test_field_checks` above with a nested array field instead.